spec/models/claim_stats/daily.rb

```diff
@@ -0,0 +1,80 @@
+require "rails_helper"
+
+RSpec.describe ClaimStats::Daily do
+  subject { ClaimStats::Daily }
+
+  describe ".to_csv" do
+    let(:started_yesterday) { [*1..3].sample }
+    let(:started_previously) { [*1..3].sample }
+
+    let(:rejected_yesterday) { [*1..3].sample }
+    let(:rejected_previously) { [*1..3].sample }
+
+    let(:approved_yesterday) { [*1..3].sample }
+    let(:approved_previously) { [*1..3].sample }
+
+    let(:yesterday) { Date.yesterday.noon }
+    let(:previously) { yesterday - 1.day }
+
+    let(:applications_started_total) {
+      started_yesterday + started_previously + rejected_yesterday + rejected_previously + approved_yesterday + approved_previously
+    }
+    let(:applications_submitted_total) {
+      rejected_previously + rejected_yesterday + approved_previously + approved_yesterday
+    }
+    let(:applications_rejected_total) {
+      rejected_previously + rejected_yesterday
+    }
+    let(:applications_accepted_total) {
+      approved_yesterday + approved_previously
+    }
+    let(:applications_started_daily) {
+      started_yesterday + rejected_yesterday + approved_yesterday
+    }
+    let(:applications_submitted_daily) {
+      rejected_yesterday + approved_yesterday
+    }
+    let(:applications_rejected_daily) {
+      rejected_yesterday
+    }
+    let(:applications_accepted_daily) {
+      approved_yesterday
+    }
+
+    before do
+      [EarlyCareerPayments, MathsAndPhysics, StudentLoans].each do |policy|
+        started_yesterday.times { create(:claim, policy: policy, created_at: yesterday) }
+        started_previously.times { create(:claim, policy: policy, created_at: previously) }
+
+        approved_yesterday.times { create(:claim, :approved, policy: policy, created_at: yesterday, submitted_at: yesterday + 60.seconds) }
+        approved_previously.times { create(:claim, :approved, policy: policy, created_at: previously, submitted_at: previously + 60.seconds) }
+
+        rejected_yesterday.times { create(:claim, :rejected, policy: policy, created_at: yesterday, submitted_at: yesterday + 60.seconds) }
+        rejected_previously.times { create(:claim, :rejected, policy: policy, created_at: previously, submitted_at: previously + 60.seconds) }
+      end
+
+      allow_any_instance_of(Decision).to receive(:readonly?).and_return(false)
```

Comment:
@d-a-v-e Can we do this without the old style of `allow_any_instance_of`?
https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

Reply:
In that instance I’m just using this so I can get the data in the correct state. decisions don’t allow us to write to them once written and I'm not using this for testing `Decision`.

Reply:
Ah, ok, thanks for the explaination :+1: